pyteal/ast/subroutine.py

```diff
@@ -49,6 +49,16 @@ def __init__(
                     )
                 )
 
+        for var, var_type in implementation.__annotations__.items():
```

Comment:
nit: I believe this line can take advantage of previously defined `sig` on line 32.

Reply:
Looking closer at it, I believe I could. Use the `sig` approach I would have query the parameters in a different way from the return so it's a bit more cumbersome. Do you see any clear advantage to changing the logic to use `sig` instead of `implementation.__annotations__` ?